File: 0214-shortest-palindrome/0214-shortest-palindrome.rb
# @param {String} s
# @return {String}
def shortest_palindrome(s)
    if s.length < 2
        return s
    end
    rev_s = s.reverse
    pre, post = s, rev_s
    while pre != post
        pre = pre[0...-1]
        post = post[1..-1]
    end
    return rev_s[0...(rev_s.length - pre.length)] + s
end

File: 0214-shortest-palindrome/0214-shortest-palindrome.ts
function shortestPalindrome(s: string): string {
  if (s.length < 2) return s
  const rev: string = s.split('').reverse().join('')
  let pre = s
  let post = rev
  while (pre !== post) {
    pre = pre.slice(0, pre.length - 1)
    post = post.slice(1)
  }
  return rev.slice(0, rev.length - pre.length) + s
}